pkg/kubectl/proxy_server.go

```diff
@@ -158,11 +158,24 @@ func NewProxyServer(filebase string, apiProxyPrefix string, staticPrefix string,
 
 // Serve starts the server (http.DefaultServeMux) on given port, loops forever.
 func (s *ProxyServer) Serve(port int) error {
+	l, err := s.Listen(port)
+	if err != nil {
+		return err
+	}
+	return s.ServeOnListener(l)
+}
+
+// Listen is a simple wrapper around net.Listen.
+func (s *ProxyServer) Listen(port int) (net.Listener, error) {
```

Comment:
Looks like `ProxyServer` contains all the information needed to start a proxy except for `port`.  How would you feel about adding `port` and making `ProxyServer` hold all the information it needs?